test/stock-out.spec.js

```diff
@@ -0,0 +1,147 @@
+//const { once } = require('events');
+const ExcelJS = require('exceljs');
+
+const { stockOutScenario, stockCountScenario } = require('./mocks/mocks');
+const { 
+  setDirToprojectConfig,
+  revertBackToProjectHome
+} = require('./test-utils');
+
+
+describe('Stock out', () => {
+  const workingDir = process.cwd();
+
+  beforeEach(() => {
+    revertBackToProjectHome(workingDir);
+    setDirToprojectConfig();
+  });
+
+  afterEach(async() => {
+    revertBackToProjectHome(workingDir);
+  });
+
+  it('Add stock out summaries test', async() => {
+    const processDir = process.cwd();
+    const workbook = new ExcelJS.Workbook();
+    const childProcess = spawnSync('../../main.js',  stockOutScenario.initScenario);
+
+    if(childProcess.status === 0) {
+      // Check that stock monitoring is initialized and stock count and stock out xform and properties files are generated
+      //const stockMonitoringConfig = path.join(processDir, 'stock-monitoring.config.json');
+
+      const formFiles = fs.readdirSync(path.join(processDir, 'forms', 'app'));
+      for(const formFile of formFiles){
+        if(formFile.toString().includes('stock_count') || formFile.toString().includes('stock_out')){
+          expect(fs.existsSync(path.join(processDir, 'forms', 'app', formFile))).toBe(true);
+        }
+      }
+
+      expect(fs.existsSync(path.join(processDir, 'stock-monitoring.config.json'))).toBe(true);
+
+      // Check that the products are available in stock count xform
+      const productCategoryList = stockCountScenario.productCategoryScenario;
+      const productsList = stockCountScenario.productsScenario;
+      await workbook.xlsx.readFile(path.join(processDir, 'forms', 'app', `stock_count.xlsx`));
+      const surveyWorkSheet = workbook.getWorksheet('survey');
+      const nameCol = surveyWorkSheet.getColumn('B');
+      const cellProductCategoriesList = [];
+      const cellProductsList = [];
+      let productIndex = 0;
+      let productCatIndex =0;
+      nameCol.eachCell(function(cell){
+
+        if(cell.value === productCategoryList[productCatIndex] && productCatIndex < productCategoryList.length){
+          cellProductCategoriesList.push(cell.value);
+          productCatIndex ++;
+          productIndex = 0;
+        }
+
+        if(cell.value === productsList[productIndex] && productIndex < productsList.length){
+          cellProductsList.push(cell.value);
+          productIndex ++;
+        }
+        
+      });
+
+      expect(productsList.length).toBe(cellProductsList.length);
+      expect(productsList.entries).toStrictEqual(cellProductsList.entries);
+      
+      expect(productCategoryList.length).toBe(cellProductCategoriesList.length);
+      expect(productCategoryList.entries).toStrictEqual(cellProductCategoriesList.entries);
+
+      // Add stock out feature test
+      const stockOutChildProcess = spawnSync('../../main.js', stockOutScenario.addStockOutFeatureScenario);
+      if(stockOutChildProcess.status === 0){
```

Comment:
```suggestion
    if (stockOutChildProcess.error) {
          throw stockOutChildProcess.error;
    }
```

better to avoid long if and throw error